samples/mvs_command_support.py

```diff
@@ -0,0 +1,214 @@
+            if len(listitem) > 72:
+                print("Input lines must be less than 72 chars\n")
+                print(f"{listitem} is length: {len(listitem)} and is ignored")
+
+            else:
+                sysin.write(f"{listitem}\n")
+
+
+def create_input_dd(input_list : list, ddname : str="SYSIN")->DDStatement:
+    """Create an input DD basedd on a list
+
+    Parameters:
+        input_list - <list>: A list of strings which is input to the DD
+        ddname - <str> (optional): The DDName used for input. Defaults to "SYSIN".
+    Return:
+        <DDStatement>: The created DD statement 
+    """
+    # First we need to create the name of the temporary file
+    # Use the ddname in the file
+    temporary_file_name = get_temp_file_name(ddname)
+
+    # Now create the file that will hold the input
+    create_input_file(input_list, temporary_file_name)
+
+    # Now create the DD that will hold the input
+
+    return DDStatement(ddname, FileDefinition(temporary_file_name))
+
+
+def cleanup_temporaries(debugging :bool=False):
+    """Remove any temporary files or datasets
+
+    Args:
+        debugging - <bool> (optional): Debug flag. If set keep files around
+        _mcs_data_set_list - <list> (implicit): Global variable that is updated
+                                                whenever a dataaet or file is created
+    Return:
+        None
+    """
+    for dataset in _mcs_data_set_list:
+        if "TEMPRARY" in dataset:
+            if "/" in dataset:
+                if debugging:
+                 print(f"Temporary file: {dataset} has not been erased")
+                else:
+                    os.remove(dataset)
+            else:
+                if debugging:
+                    print(f"Temporary Dataset: {dataset} has not been erased")
+                else:
+                    # All uppercase names are DATASETS
+                    return_code = datasets.delete(dataset)
+                    if return_code != 0:
+                        print(f"Error erasing {dataset}")
+
+def main():
+    """Test this with a SMPE list
+    """
+    # First define a list of DD statements
+    dd_list=[]
+
+    # Create an input dd to list the Global Zone
+    dd_list.append(create_input_dd([" SET BDY(GLOBAL)."," LIST. "],ddname="SMPCNTL"))
+
+    # Define the Workspace dataset
+    workspace_dataset={"type":"PDS","primary_space":"1G","secondary_space":"1G",
+                       "block_size":3200,"record_format":"FB","record_length":80,
+                       "volumes":"USRAT8"}
+    workspace_dataset=create_temp_dataset(workspace_dataset)
+
+    # Add it to the DD list
+    dd_list.append(DDStatement("SMPWRK6",DatasetDefinition(workspace_dataset["name"])))
+
+    # Define the Dummy DDs
+    dd_list.append(DDStatement("SMPLOG","DUMMY"))
+    dd_list.append(DDStatement("SMPLOGA","DUMMY"))
+
+    # Define the Global CSI
+    dd_list.append(DDStatement("SMPCSI",DatasetDefinition("AT4SMP.GLOBAL.CSI")))
```

Comment:
This dataset name looks rather specific.  Is there a better way than hardcoding it?

Reply:
We should discuss this. I used what is in the main for my testing. The dataset name is not the only thing that is hard coded. Does it make sense to build a more complex main that takes parameters for all of the hard coded items - Or is it smarter to build a comment section that describes it better. The idea is to get people to use each of the functions rather than use them as part of the main.